src/terraform/modules/controller/main.tf

```diff
@@ -141,31 +145,31 @@ locals {
 }
 
 resource "azurerm_role_assignment" "avere_create_cluster_role" {
-  count                            = length(local.avere_contributor_rgs)
+  count                            = var.deploy_controller ? length(local.avere_contributor_rgs) : 0
   scope                            = "${data.azurerm_subscription.primary.id}/resourceGroups/${local.avere_contributor_rgs[count.index]}"
   role_definition_name             = local.avere_create_cluster_role
-  principal_id                     = azurerm_linux_virtual_machine.vm.identity[0].principal_id
+  principal_id                     = azurerm_linux_virtual_machine.vm[0].identity[0].principal_id
```

Comment:
I'm assuming the resource "azurerm_linux_virtual_machine" returns an array due to the addtion of the 'count' parameter to the resource?

Reply:
correct, once the "count" is added to a resource, it automatically converts it to a list